### PR TITLE
fpgad: add systemd integration

### PR DESCRIPTION
**Pull request by tswhison**

### CODE REVIEW

File: tools/base/fpgad/fpgad.service.in [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Intel FPGA daemon
+[Unit]
+Description = FPGA monitor
+
+[Service]
+Type=simple
+StandardOutput=journal+console
+StandardError=journal+console
+EnvironmentFile=-/etc/sysconfig/fpgad.conf
```
Comment (zhangzhang10) [inline tools/base/fpgad/fpgad.service.in]: I'm not familiar with the syntax of the service config file. But is the **=-** intentional and correct for EnvironmentFile?

Comment (asgibbons) [reply tools/base/fpgad/fpgad.service.in]: The '-' after the '=' says to ignore the error if the file does not exist.

File: tools/base/fpgad/CMakeLists.txt [hunk 2]
```diff
@@ -62,12 +62,18 @@ target_link_libraries(fpgad
...
 
 if ("${CMAKE_INSTALL_PREFIX}" STREQUAL "/usr")
   install(FILES fpgad.cfg DESTINATION /etc/opae COMPONENT toolfpgad)
+  install(FILES fpgad.conf DESTINATION /etc/sysconfig COMPONENT toolfpgad)
 else()
   install(FILES fpgad.cfg DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/opae COMPONENT toolfpgad)
+  install(FILES fpgad.conf DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/sysconfig COMPONENT toolfpgad)
 endif()
+
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/fpgad.service DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/systemd/system COMPONENT toolfpgad)
```
Comment (asgibbons) [inline tools/base/fpgad/CMakeLists.txt]: Have you tested this?  I was unable to get the files to install automatically.  I think the "real" path is /etc/systemd/system

Comment (tswhison) [reply tools/base/fpgad/CMakeLists.txt]: I did, yes..  there was a problem with the definition of CMAKE_INSTALL_PREFIX.  The tools directory was being traversed before the variable had been set.  Thanks to @nakulkorde for noticing the issue.  Ok, I will look into changing it to /etc/systemd/system.  Thanks!

Comment (asgibbons) [reply tools/base/fpgad/CMakeLists.txt]: OK.  Just to be clear, the service files are in /lib/systemd/system for Ubuntu, but /etc/systemd/system for CentOS/RHEL.  I believe this was one sticky point I had with my implementation.

I also meant that you need to install the .rpd or .deb files on a clean system to see whether they got installed in the proper place, and the '@' substitution put in the correct paths.  You may have done that...

Comment (tswhison) [reply tools/base/fpgad/CMakeLists.txt]: Yep, I verified that the files were going to the right spot.
Re: the target directory for the .service, I checked on one of our RHEL 7.6 boxes, and it looks like /usr/lib/systemd/system is the right place.  I see several other .service files there.  Seem OK?

Comment (asgibbons) [reply tools/base/fpgad/CMakeLists.txt]: Look [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/chap-Managing_Services_with_systemd#tabl-Managing_Services_with_systemd-Introduction-Units-Locations).  On the minicloud machine DCP uses for testing (CentOS 7) it is in /etc/systemd/system.

I believe we need to 'systemctl enable' these services, which implies (from the table) that they belong in /etc.

Comment (asgibbons) [reply tools/base/fpgad/CMakeLists.txt]: In thinking about this, I believe the current approach - installing in opae directory and requiring admin to edit and move them - makes the most sense.  These files will be in different places depending on the whim of the admin.  system.conf can redirect these directories anywhere one wants.

Comment (tswhison) [reply tools/base/fpgad/CMakeLists.txt]: Ok, good point.  Let me seek guidance on the one-click requirement.

Comment (tswhison) [reply tools/base/fpgad/CMakeLists.txt]: I was told the preference was to install the files in their final destinations.

**Pull request merged**